InnerEye/ML/lightning_models.py

```diff
@@ -133,11 +136,18 @@ def compute_metrics(self, cropped_sample: CroppedSample, segmentation: torch.Ten
             self.storing_logger.train_diagnostics.append(center_indices)
         else:
             self.storing_logger.val_diagnostics.append(center_indices)
-        # if self.train_val_params.in_training_mode:
-        #     # store the sample train patch from this epoch for visualization
-        #     if batch_index == self.example_to_save and self.config.store_dataset_sample:
-        #         _store_dataset_sample(self.config, self.train_val_params.epoch, forward_pass_result,
-        #                               cropped_sample)
+
+        if is_training and self.config.store_dataset_sample:
+            # store the sample train patch from this epoch for visualization
+            # remove batches and channels
+            dataset_example = DatasetExample(image=cropped_sample.image[0][0].cpu().detach().numpy(),
```

Comment:
In practice, did that provide a good amount of insight? You would get a different subject in each epoch.